### ISSUE TITLE
'az' not found on PATH, please make sure it is installed

### ISSUE THREAD

**Issue by wshamroukh** [microsoft/vscode-azurecli#102]
I have been trying to fix this issue now for a while. I tried everything, uninstall/reinstalling Azure CLI tools extension, uninstalling/reinstalling vscode, but whenever I create a new .azcli or open an existing one, I get this message

![image](https://github.com/user-attachments/assets/d4e253e7-b8a4-49ff-b1e4-9532aab1c21a)

az cli is already in path on all terminals:

![image](https://github.com/user-attachments/assets/2c3c0a90-3d77-4f7b-a1e8-70b826f89442)

![image](https://github.com/user-attachments/assets/63a64e94-3aa7-476a-b48a-b1fe091643bb)

![image](https://github.com/user-attachments/assets/0e70cd77-2106-4baa-a9da-355039686074)

In vscode Developer Tools these errors are logged while trying to activate the azure cli tools extension - attached the log file for your reference as well:

![image](https://github.com/user-attachments/assets/9f3a1e76-c340-4c21-bfb5-bef82700d002)

I am stuck and I don't know how to go about this issue. Can you please help?
[vscode-app-1723701074763.log](https://github.com/user-attachments/files/16621870/vscode-app-1723701074763.log)

**Comment by Kivayan** [comment]
I am facing exactly the same issues. Tried reinstalls of azcli, reinstall of extensions, removing and reading to PATH, with restarts, nothing helps. AZ works normaly from PS terminal, but this error breaks intellisense for azcli files.

**Comment by wshamroukh** [comment]
> I am facing exactly the same issues. Tried reinstalls of azcli, reinstall of extensions, removing and reading to PATH, with restarts, nothing helps. AZ works normaly from PS terminal, but this error breaks intellisense for azcli files.

Yeah the thing is that it happened out of the blue and I did everything but still the azure cli tools extension is not able to see az and hence intellisense is not working at all

**Comment by wshamroukh** [comment]
> I am facing exactly the same issues. Tried reinstalls of azcli, reinstall of extensions, removing and reading to PATH, with restarts, nothing helps. AZ works normaly from PS terminal, but this error breaks intellisense for azcli files.

It turned out to be a vscode issue - specifically the latest **version 1.92.1**
I uninstalled it and then installed **version 1.90.0** and I don't have this issue anymore - give it a try @Kivayan 
system: https://update.code.visualstudio.com/1.90.0/win32-x64/stable
user: https://update.code.visualstudio.com/1.90.0/win32-x64-user/stable

**Comment by shaneahenderson** [comment]
I have this issue both on Windows and Mac.

**Comment by wshamroukh** [comment]
> I have this issue both on Windows and Mac.

Check my previous comment

**Comment by caleb-terry** [comment]
Same issue as described. Azure CLI installed, resolves when I execute in terminal but using Azure CLI Tools results in 'az' not found on PATH, please make sure it is installed.

Version: 1.93.0 (system setup)
Commit: 4849ca9bdf9666755eb463db297b69e5385090e3
Date: 2024-09-04T13:02:38.431Z
Electron: 30.4.0
ElectronBuildId: 10073054
Chromium: 124.0.6367.243
Node.js: 20.15.1
V8: 12.4.254.20-electron.0
OS: Windows_NT x64 10.0.22631

**Comment by lysacor** [comment]
I am encountering this same issue on version 0.5.0 and 0.6.0:

Version: 1.93.1 (user setup)
Commit: 38c31bc77e0dd6ae88a4e9cc93428cc27a56ba40
Date: 2024-09-11T17:20:05.685Z
Electron: 30.4.0
ElectronBuildId: 10073054
Chromium: 124.0.6367.243
Node.js: 20.15.1
V8: 12.4.254.20-electron.0
OS: Windows_NT x64 10.0.19045

I have uninstalled and reinstalled azure-cli and checked the user and system path environment variable and I have hard coded the path there. The command prompt on the terminal application and within VSCode are able to find and run Azure CLI without issue.

**Comment by zhoxing-ms** [comment]
This issue should be caused by this breaking change of VS code https://code.visualstudio.com/updates/v1_92#_electron-30-update
You can refer to my two commits  https://github.com/zhoxing-ms/vscode-azurecli/commit/d1f85e39dc27b3ed4e1945920122f0c3eb09baee and https://github.com/zhoxing-ms/vscode-azurecli/commit/53bc4402bc9d0b7748335dee57826c3f6dfc8734 to solve this issue

**Comment by wshamroukh** [comment]
> This issue should be caused by this breaking change of VS code https://code.visualstudio.com/updates/v1_92#_electron-30-update You can refer to my two commits [zhoxing-ms@d1f85e3](https://github.com/zhoxing-ms/vscode-azurecli/commit/d1f85e39dc27b3ed4e1945920122f0c3eb09baee) and [zhoxing-ms@53bc440](https://github.com/zhoxing-ms/vscode-azurecli/commit/53bc4402bc9d0b7748335dee57826c3f6dfc8734) to solve this issue

So how to get this fix incorporated into az cli tools extension?

**Comment by wshamroukh** [comment]
> This issue should be caused by this breaking change of VS code https://code.visualstudio.com/updates/v1_92#_electron-30-update You can refer to my two commits [zhoxing-ms@d1f85e3](https://github.com/zhoxing-ms/vscode-azurecli/commit/d1f85e39dc27b3ed4e1945920122f0c3eb09baee) and [zhoxing-ms@53bc440](https://github.com/zhoxing-ms/vscode-azurecli/commit/53bc4402bc9d0b7748335dee57826c3f6dfc8734) to solve this issue

Thank you. It seems to work just fine now with this fix
I have applied the fix in the file `C:\Users\myusername\.vscode\extensions\ms-vscode.azurecli-0.6.0\out\src\azService.js`

**Comment by Fix9000** [comment]
> This issue should be caused by this breaking change of VS code https://code.visualstudio.com/updates/v1_92#_electron-30-update You can refer to my two commits [zhoxing-ms@d1f85e3](https://github.com/zhoxing-ms/vscode-azurecli/commit/d1f85e39dc27b3ed4e1945920122f0c3eb09baee) and [zhoxing-ms@53bc440](https://github.com/zhoxing-ms/vscode-azurecli/commit/53bc4402bc9d0b7748335dee57826c3f6dfc8734) to solve this issue

Worked for me to. Thank you.

**Comment by rosatownsend** [comment]
This fix also worked for me, applying the changes to `azService.js`

It would be nice to get it released into the extension proper.

**Comment by gihanw80** [comment]
works 👍🏽

**Comment by SitraKarel** [comment]
A year later and I still have this issue, is there any progress on this?
